source/postprocess/max_depth_field.cc

```diff
@@ -0,0 +1,131 @@
+/*
+  Copyright (C) 2021 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/max_depth_field.h>
+#include <aspect/geometry_model/interface.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    MaxDepthField<dim>::execute (TableHandler &statistics)
+    {
+      // create a quadrature formula based on the compositional element alone.
+      // be defensive about determining that a compositional field actually exists
+      AssertThrow(this->introspection().base_elements.compositional_fields != numbers::invalid_unsigned_int,
+                  ExcMessage("This postprocessor cannot be used without compositional fields."));
+      const QGauss<dim> quadrature_formula(this->get_fe().base_element(this->introspection().base_elements.compositional_fields).degree + 1);
+
+      const unsigned int n_q_points = quadrature_formula.size();
+
+      FEValues<dim> fe_values (this->get_mapping(),
+                               this->get_fe(),
+                               quadrature_formula,
+                               update_values   |
+                               update_quadrature_points |
+                               update_JxW_values);
+
+      std::vector<double> compositional_values(n_q_points);
+      std::vector<Point<dim> > position_values(n_q_points);
+
+      std::vector<double> local_max_depth(this->n_compositional_fields(), std::numeric_limits<double>::min());
```

Comment:
It is confusingly named, but `std::numeric_limits<double>::min()` is actually `1e-16` or thereabouts -- the smallest *positive* number representable in floating point, see https://en.cppreference.com/w/cpp/types/numeric_limits/min You probably want to use 
```suggestion
      std::vector<double> local_max_depth(this->n_compositional_fields(), std::numeric_limits<double>::lowest());
```
See also #4135.